src/components/vm/consoles/vnc.tsx

```diff
@@ -681,11 +686,17 @@ export const VncPending = ({
     inactive_vnc: VMGraphics,
     isExpanded: boolean,
 }) => {
+    const Dialogs = useDialogs();
+
     return (
         <>
             <EmptyState icon={PendingIcon} status="custom">
                 <EmptyStateBody>
-                    {_("Restart this virtual machine to access its graphical console")}
+                    <TextWithAction
+                        text={_("$0 this virtual machine to access its graphical console")}
+                        action={_("Restart")}
+                        onClick={() => Dialogs.show(<VmRestartDialog vm={vm} />)}
```

Comment:
This added line is not executed by any test.  [Details](https://logs-cockpit.apps.ocp.cloud.ci.centos.org/pull-2376-a00f515a-20251027-132245-fedora-42-devel/Coverage/lcov/github-pr.diff.gcov.html)